src/game/puppycam2.c

```diff
@@ -478,6 +475,80 @@ void puppycam_reset_values(void) {
+        diff[2] = ABS(newcam_fixedcam[i].newcam_hard_Z1 - newcam_fixedcam[i].newcam_hard_Z2);
+        vec3s_copy(sPuppyVolumeStack[gPuppyVolumeCount]->pos, pos);
+        vec3s_copy(sPuppyVolumeStack[gPuppyVolumeCount]->radius, diff);
+        sPuppyVolumeStack[gPuppyVolumeCount]->rot = 0; // Doesn't support rotation.
+        sPuppyVolumeStack[gPuppyVolumeCount]->func = newcam_fixedcam[i].newcam_hard_script;
+        tempAngle->yaw = PUPPY_NULL;
+        tempAngle->pitch = PUPPY_NULL;
+        tempAngle->zoom = PUPPY_NULL;
+        tempAngle->pos[0] = newcam_fixedcam[i].newcam_hard_camX;
+        tempAngle->pos[1] = newcam_fixedcam[i].newcam_hard_camY;
+        tempAngle->pos[2] = newcam_fixedcam[i].newcam_hard_camZ;
+        tempAngle->focus[0] = newcam_fixedcam[i].newcam_hard_lookX;
+        tempAngle->focus[1] = newcam_fixedcam[i].newcam_hard_lookY;
+        tempAngle->focus[2] = newcam_fixedcam[i].newcam_hard_lookZ;
+        sPuppyVolumeStack[gPuppyVolumeCount]->angles = tempAngle;
+        switch (newcam_fixedcam[i].newcam_hard_modeset) {
+            case NC_MODE_FIXED:
+                flagsAdd = 0;
+                flagsRemove = PUPPYCAM_BEHAVIOUR_TURN_HELPER | PUPPYCAM_BEHAVIOUR_INPUT_NORMAL | PUPPYCAM_BEHAVIOUR_INPUT_4DIR |
+                PUPPYCAM_BEHAVIOUR_INPUT_8DIR | PUPPYCAM_BEHAVIOUR_INPUT_2D;
+                break;
+            case NC_MODE_FIXED_NOTURN:
+                flagsAdd = 0;
+                flagsRemove = PUPPYCAM_BEHAVIOUR_PITCH_ROTATION | PUPPYCAM_BEHAVIOUR_YAW_ROTATION |
+                PUPPYCAM_BEHAVIOUR_TURN_HELPER | PUPPYCAM_BEHAVIOUR_INPUT_NORMAL | PUPPYCAM_BEHAVIOUR_INPUT_4DIR |
+                PUPPYCAM_BEHAVIOUR_INPUT_8DIR | PUPPYCAM_BEHAVIOUR_INPUT_2D;
+                break;
+            case NC_MODE_NOROTATE:
+                flagsAdd = 0;
+                flagsRemove = PUPPYCAM_BEHAVIOUR_PITCH_ROTATION | PUPPYCAM_BEHAVIOUR_YAW_ROTATION;
+                break;
+            case NC_MODE_FIXED_NOMOVE:
+                flagsAdd = 0;
+                flagsRemove = PUPPYCAM_BEHAVIOUR_X_MOVEMENT | PUPPYCAM_BEHAVIOUR_Y_MOVEMENT | PUPPYCAM_BEHAVIOUR_Z_MOVEMENT;
+                break;
+            default:
+                flagsAdd = 0;
+                flagsRemove = 0;
+                break;
+        }
+        sPuppyVolumeStack[gPuppyVolumeCount]->flagsAdd = flagsAdd;
+        sPuppyVolumeStack[gPuppyVolumeCount]->flagsRemove = flagsRemove;
+        sPuppyVolumeStack[gPuppyVolumeCount]->flagPersistance = newcam_fixedcam[i].newcam_hard_permaswap;
+        sPuppyVolumeStack[gPuppyVolumeCount]->shape = PUPPYVOLUME_SHAPE_BOX;
+        sPuppyVolumeStack[gPuppyVolumeCount]->room  = -1;
+        sPuppyVolumeStack[gPuppyVolumeCount]->fov  = 45;
+        sPuppyVolumeStack[gPuppyVolumeCount]->area  = newcam_fixedcam[i].newcam_hard_areaID;
+        gPuppyVolumeCount++;
+        append_puppyprint_log("X: %d, Y: %d, Z: %d", pos[0], pos[1], pos[2]);
```

Comment:
this shouldn't pollute the logs, if it's a convenient log to have in here i would comment it out or have a define or macro at the top of this file for cam debug

Reply:
oops dev debugging leftover